core/shared/platform/common/posix/posix_memmap.c

```diff
@@ -5,7 +5,8 @@
 
 #include "platform_api_vmcore.h"
 
-#if (defined(__APPLE__) || defined(__MACH__)) && defined(__arm64__)
+#if (defined(__APPLE__) || defined(__MACH__)) && defined(__arm64__) \
+    && TARGET_OS_OSX
```

Comment:
Using TARGET_OS_OSX may cause compilation error in some platforms, do you mean `defined(TARGET_OS_OSX)`?

Reply:
iOS and MacOS both defined TARGET_OS_OSX. 
MacOS:
#define TARGET_OS_OSX               1
iOS:
#define TARGET_OS_OSX               0

Reply:
"Defined (_ APPLE__) | | defined (_ MACH__) " has already indicated that it is the Apple platform, and all Apple platforms have defined TARGET_OS_OSX

Reply:
So seems it had better be `&& defined(TARGET_OS_OSX) && TARGET_OS_OSX != 0`?

Reply:
it looks better now！！！

Reply:
> "defined (__ APPLE__) | | defined (__ MACH__) " has already indicated that it is the Apple platform, and all Apple platforms have defined TARGET_OS_OSX

OK, if that, `TARGET_OS_OSX != 0` should be good.

Another question is should we apply similar change to sys_icache_invalidate? Is this API also only defined in macOS?

Reply:
Sorry, I made a mistake. This method is defined on both iOS and macOS and has no restrictions on ARM architecture. 
It would be better to change it to the following：
```
#if (defined(__APPLE__) || defined(__MACH__))
#include <libkern/OSCacheControl.h>
#endif
```
```
f (defined(__APPLE__) || defined(__MACH__))
    sys_icache_invalidate(start, len);
#endif
```

Reply:
Thanks, sounds good.